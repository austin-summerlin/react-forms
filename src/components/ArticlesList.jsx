/* eslint-disable keyword-spacing */
import React from 'react';
import PropTypes from 'prop-types';
import Article from './Article';

const ArticlesList = ({ articles }) => {
  if (articles) {
    const articleElements = articles.map(article => (
      <li
        key={article.title}
        style={{
          listStyle: 'none',
          fontSize: '.7em',
          border: '2px solid black',
          borderRadius: '5px',
          margin: '2px',
          backgroundColor: 'lightpink'
        }}>
        <Article
          title={article.title}
          author={article.author}
          url={article.url}
        />
      </li>
    ));
    return <ul>{articleElements}</ul>;
  } return <ul>Oops...something went wrong</ul>;
};

ArticlesList.propTypes = {
  articles: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.stringisRequired,
      author: PropTypes.string,
      url: PropTypes.string
    })
  ),
};

export default ArticlesList;
