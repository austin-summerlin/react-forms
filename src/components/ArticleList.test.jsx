/**
 * @jest-environment jsdom
 */
import React from 'react';
import { render } from '@testing-library/react';
import ArticlesList from './ArticlesList';

describe('ArticlesList component', () => {
  it('should render component', () => {
    const articles = [{
      title: 'test',
      author: 'test',
      url: 'test',
    }];

    const { asFragment } = render(<ArticlesList articles={articles} />);

    expect(asFragment).toMatchSnapshot();
  });
});
