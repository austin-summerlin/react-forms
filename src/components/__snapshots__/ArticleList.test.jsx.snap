// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticlesList component should render component 1`] = `[Function]`;
